### PR TITLE
QR Code Generator

### PR DESCRIPTION
**Pull request by chinazor-allen**
#### What does this PR do?
Allows users to generate QR Code of inputed text

#### Description of Task to be completed?
 -  As a developer, I want to be able to generate data for creating QR Code Image

#### What are the relevant JIRA stories?
`SL-37`

### CODE REVIEW

File: test/generateQRCodeTest.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const meta = generateMeta('generate-qr-code-image');
+let args = {};
+
+describe('Generate QR Code', () => {
+  it('returns an error if a `GET` request method is passed to Url', (done) => {
+    meta.request.REQUEST_METHOD = 'GET';
+    run('generate-qr-code-image', {
+      meta
+    }).then((response) => {
+      expect(response.data.message).to.equal('Make sure to use `POST`' +
```
Comment (adamwardecki) [inline test/generateQRCodeTest.js]: use template literals or assign to var instead of string concatenation

File: socket.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: qr-code
+description: QR Code Generator Integration
+version: 0.0.1
+
+endpoints:
+  generate-qr-code-image:
+    description: Generates qr code image of text inputed.
```
Comment (adamwardecki) [inline socket.yml]: Generates QR code image from an input text

File: src/generate-qr-code-image.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Syncano from 'syncano-server';
+import qr from 'qr-image';
+
+export default async (ctx) => {
+  const { response } = Syncano(ctx);
+
+  const checkQRImage = () => new Promise((resolve, reject) => {
+    if (!ctx.args.text) {
+      return reject(response.json({
```
Comment (adamwardecki) [inline src/generate-qr-code-image.js]: you are using the `response.json()` method in the catch clause below. You can reject with just a message that will be handled in the try catch.

**Review by adamwardecki — CHANGES_REQUESTED** [review]
Looks good! Only minor remarks, please see the comments.

File: socket.yml (hunk 2 above)
Comment (chinazor-allen) [reply socket.yml]: fixed

File: src/generate-qr-code-image.js (hunk 3 above)
Comment (chinazor-allen) [reply src/generate-qr-code-image.js]: fixed

File: test/generateQRCodeTest.js (hunk 1 above)
Comment (chinazor-allen) [reply test/generateQRCodeTest.js]: fixed

**Review by adamwardecki — APPROVED** [review]

**Pull request merged**